desktop-widgets/diveplanner.cpp

```diff
@@ -563,17 +563,20 @@ void PlannerWidgets::planDive()
 
 void PlannerWidgets::replanDive()
 {
+	if (!current_dive)
+		return;
+	copy_dive(current_dive, &displayed_dive); // Planning works on a copy of the dive (for now).
 	DivePlannerPointsModel::instance()->setPlanMode(DivePlannerPointsModel::PLAN);
 
 	MainWindow::instance()->graphics->setPlanState();
 
 	plannerWidget.setReplanButton(true);
-	plannerWidget.setupStartTime(timestampToDateTime(current_dive->when));
-	if (current_dive->surface_pressure.mbar)
-		plannerWidget.setSurfacePressure(current_dive->surface_pressure.mbar);
-	if (current_dive->salinity)
-		plannerWidget.setSalinity(current_dive->salinity);
-	DivePlannerPointsModel::instance()->loadFromDive(current_dive);
+	plannerWidget.setupStartTime(timestampToDateTime(displayed_dive.when));
+	if (displayed_dive.surface_pressure.mbar)
+		plannerWidget.setSurfacePressure(displayed_dive.surface_pressure.mbar);
+	if (displayed_dive.salinity)
+		plannerWidget.setSalinity(displayed_dive.salinity);
+	DivePlannerPointsModel::instance()->loadFromDive(&displayed_dive);
```

Comment:
This needs a closer review. `current_dive` is usually a dive in the dive list.
Whereas `displayed_dive` is a copy of such a dive. So without more searching (which I don't have time for) this seems like it changes behavior?

Reply:
Note that a few lines above `current_dive` is copied into `displayed_dive`, so there should be no behavioral change.

Once the profile can display arbitrary dives, the `displayed_dive` will be made a _local_ copy of the dive-to-be-edited. Ultimately, it should be removed completely with fine-grained undo.